### DISCUSSION TITLE
Generate within selection while ignoring context

### DISCUSSION THREAD

**Discussion by Smorgin** [Acly/krita-ai-diffusion#661] [Ideas]
It seems odd to me that the only option for a completely fresh image creation is tied to full canvas size. I thought that "focus" under the custom generation options would allow this but even enabled and at 100% strength it will still try to match within context. 
## Example of "a cat and red apple" on a white bg
![krita_2rEgoaYs8b](https://github.com/Acly/krita-ai-diffusion/assets/168103343/3921cfc1-5d4b-491e-b8aa-623594ac8b32)
While generally the context matching is helpful I would like to be able to turn it off and generate freshly within a selection via the custom generate.

**Comment by Acly** [comment]
That's possible, make sure to turn off "Seamless" and "Focus" and choose "Selection Bounds" as context.
Check out https://github.com/Acly/krita-ai-diffusion/wiki/Inpainting#custom-generation if you haven't already.

**Reply by qrazil** [reply]
Hey, recently started using this plugin and noticed that using selection with seamless and focus turned off on an empty canvas creates a desaturated output. Guessing it is using the empty canvas to vaeencode. Can we add an option 'from scratch' or something similar thats allows user to generate completely new image using an empty latent in selection? 

Usecase: manga/comic generation using ai, requires generation in panels. Gets complicated if requires global mask selection. Also this alternative feels more intuitive. Let me know what you think. 

I could work on the PR if needed.

**Reply by Acly** [reply]
You might have to turn off "Color match" in diffusion settings, it's probably causing the desaturation because there is no color outside your selection.

I think it can be improved to detect that case. See also #2483

**Reply by qrazil** [reply]
Hey, that works perfectly! Would it be better to add that to selection menu? Feels a bit cumbersome to switch to settings and back, if user wants to generate then refine, and switch quickly in between? Is there some reason to keep that in the settings? Does it affect any other mode?

**Reply by Acly** [reply]
Currently it only works on selections, but it would be good if it didn't (ie. there was a robust heuristic to detect which things were supposed to change vs. which weren't after an edit, without an explicit selection).

It might make sense to add an alternative to Fill/Expand/etc. modes that reflects the image-within-the-image use case that you get with comic panels.

**Reply by qrazil** [reply]
The heuristic option might be difficult to create and maintain if different users have different expectations. Might be easier to just give control over to user. The selection is current best option. So, should we move 'color match' to selection menu? If we come up with a heuristic later that works for multiple tools, we can add that to the diffusion setting.

**Reply by qrazil** [reply]
Should I raise a PR to move that setting to selection menu then?

**Reply by Acly** [reply]
Selection menu toggle is not a good solution, I'd prefer a separate mode (like Fill/Expand/etc.) for this use case. It's several settings that are affected (inpaint model, fill mode, context) not just color match.

**Reply by qrazil** [reply]
Since it matches/applies for multiple modes, for example fill/expand/etc, shouldn't color match be a checkbox that applies/does not apply regardless of mode?
